Content.Server/Devour/DevourSystem.cs

```diff
@@ -1,49 +1,89 @@
+        //Store the target in the stomach if it's allowed.
+        if (component.ShouldStoreDevoured && HasComp<MobStateComponent>(args.Args.Target))
+        {
+            ContainerSystem.Insert(args.Args.Target.Value, component.Stomach);
+
+            EnsureComp<DevouredComponent>(args.Args.Target.Value);
+
+            var ev = new DevouredEvent(uid);
+            RaiseLocalEvent(args.Args.Target.Value, ref ev);
+        }
 
+        //Only alter the passive healing if the devoured entity belongs to the devourers food preference.
         if (component.FoodPreference == FoodPreference.All ||
-            (component.FoodPreference == FoodPreference.Humanoid && HasComp<HumanoidAppearanceComponent>(args.Args.Target)))
+            (component.FoodPreference == FoodPreference.Humanoid &&
+             HasComp<HumanoidAppearanceComponent>(args.Args.Target)))
         {
-            ichorInjection.ScaleSolution(0.5f);
-
-            if (component.ShouldStoreDevoured && args.Args.Target is not null)
+            if(TryComp<PassiveDamageComponent>(uid, out var passiveHealing))
             {
-                ContainerSystem.Insert(args.Args.Target.Value, component.Stomach);
+                if (component.PassiveDevourHealing != null)
+                {
+                    passiveHealing.Damage += component.PassiveDevourHealing;
+                }
             }
-            _bloodstreamSystem.TryAddToChemicals(uid, ichorInjection);
         }
 
         //TODO: Figure out a better way of removing structures via devour that still entails standing still and waiting for a DoAfter. Somehow.
-        //If it's not human, it must be a structure
+        //If it does not have a mobState, it must be a structure
         else if (args.Args.Target != null)
         {
             QueueDel(args.Args.Target.Value);
         }
 
-        _audioSystem.PlayPvs(component.SoundDevour, uid);
+        AudioSystem.PlayPvs(component.SoundDevour, uid);
+    }
+
+    /// <summary>
+    /// Removes the devoured component from entities in stomach so it's effects can be removed.
+    /// Empties the stomach so the entities in it don't get deleted together with the dragon.
+    /// </summary>
+    public void EmptyStomach(EntityUid uid, DevourerComponent component)
+    {
+        foreach (var entity in component.Stomach.ContainedEntities)
+        {
+            RemComp<DevouredComponent>(entity);
+        }
+        ContainerSystem.EmptyContainer(component.Stomach);
     }
-}
 
+    /// <summary>
+    /// Empties the stomach when gibbed.
+    /// </summary>
+    private void OnBeingGibbed(EntityUid uid, DevourerComponent component, BeingGibbedEvent args)
+    {
+        EmptyStomach(uid, component);
+    }
+
+    /// <summary>
+    /// Empties the stomach when component is removed.
+    /// </summary>
+    private void OnRemoved(EntityUid uid, DevourerComponent component, ComponentRemove args)
```

Comment:
i could be misremembering but i could swear gibbing raised ComponentRemove since it deletes the entity, is the other sub necessary?